Framework/MDAlgorithms/src/IntegrateQLabEvents.cpp

```diff
@@ -301,6 +314,25 @@ PeakShapeEllipsoid_const_sptr IntegrateQLabEvents::ellipseIntegrateEvents(
   inti = peak_w_back.first - ratio * backgrd.first;
   sigi = sqrt(peak_w_back.second + ratio * ratio * backgrd.second);
 
+  if (inti < 0) {
+    std::ostringstream msg;
+    msg << "Negative intensity found: " << inti << "\n"
+        << "You might want to adjust the size of peak and background shell.\n";
```

Comment:
Is there any way we can give the user more specific information such as "You might want to increase the background shell thickness" or "decrease the peak radius"

Reply:
I don't really see a way where we can advise the user to pick one over another based on the numbers we have. It is better to let the user decide which way is physically meaningful.